MonoBehaviourValidated.cs

```diff
@@ -0,0 +1,14 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace KBCore.Refs
+{
+    public class MonoBehaviourValidated : MonoBehaviour

```

Comment:
Oh nice I like the idea of including this, just a couple thoughts: 

- `OnValidate` should probably be `protected virtual void OnValidate` so it can still be extended
- `OnValidate` also needs to be wrapped in `#if UNITY_EDITOR` as `ValidateRefs` is only available in editor and not in builds
- (really minor) what do you think of naming this `ValidatedMonoBehaviour`? I know it's personal preference but to me it just reads better having the adjective in front.